src/services/offer.js

```diff
@@ -152,14 +142,9 @@ class OfferService {
             query,
             {
                 isHidden: false,
-                $unset: { hiddenReason: undefined, adminReason: undefined }, // Removing property from document.
+                $unset: { hiddenReason: "", adminReason: "" }, // Removing property from document.
```

Comment:
Shouldn't these be set to null instead of an empty string?

Reply:
Since the purpose is to have them removed from the document itself, and not just "nullify" their value

Reply:
https://www.mongodb.com/docs/manual/reference/operator/update/unset/#mongodb-update-up.-unset

The specified value doesn't impact the operation, "" is used in the examples. I'm just not sure as to why `undefined` stopped working, I'm yet to find the change on that

Reply:
https://mongoosejs.com/docs/migrating_to_6.html#removed-omitundefined

Reply:
Doesn't seem like it. That's $set instead of $unset, and the values weren't being removed, they were keeping their old values for some reason

Anyway, now I'm trying to fix the remaining 10 tests but have yet to find the cause, something to do with dependencies with previous tests... We really need #169

Reply:
Gonna bump the priority on that one then

Reply:
Also, just to clarify on the subject, unless I misunderstood something, the values were keeping their old value because when using `undefined` those keys were being removed from the update payload and this not update anything in actuality.

Not that it matters much if the current version works as intended unless there are some new "best practices" regarding this matter.

Reply:
Yes I think you're completely right!